tests/unit/s2n_async_pkey_test.c

```diff
@@ -142,6 +144,43 @@ int async_pkey_store_callback(struct s2n_connection *conn, struct s2n_async_pkey
     return S2N_SUCCESS;
 }
 
+struct task_params {
+    struct s2n_connection *conn;
+    struct s2n_async_pkey_op *op;
+};
+
+void *pkey_task(void *param)
+{
+    struct task_params *params = (struct task_params*)param;
+
+    struct s2n_cert_chain_and_key *chain_and_key = s2n_connection_get_selected_cert(params->conn);
+    EXPECT_NOT_NULL(chain_and_key);
+
+    s2n_cert_private_key *pkey = s2n_cert_chain_and_key_get_private_key(chain_and_key);
+    EXPECT_NOT_NULL(pkey);
+
+    EXPECT_SUCCESS(s2n_async_pkey_op_perform(params->op, pkey));
+    EXPECT_SUCCESS(s2n_async_pkey_op_apply(params->op, params->conn));
+
+    free(params);
+    pthread_exit(NULL);
+}
+
+int async_pkey_perform_op(struct s2n_connection *conn, struct s2n_async_pkey_op *op)
+{
+    EXPECT_NOT_NULL(conn);
+    EXPECT_NOT_NULL(op);
+
+    struct task_params *params = malloc(sizeof(struct task_params));
+    params->conn = conn; 
+    params->op = op; 
+
+    pthread_t worker;
+    POSIX_GUARD(pthread_create(&worker, NULL, &pkey_task, params));
+
+    return S2N_SUCCESS;
```

Comment:
Should we worry about this test being potentially flaky, due to https://github.com/aws/s2n-tls/issues/2757?

Reply:
Yes, I did not see a way we could modify the async state directly